language/compiler/ir_to_bytecode/syntax/src/ast.rs

```diff
@@ -122,6 +122,8 @@ pub struct StructDefinition {
     pub is_nominal_resource: bool,
     /// Human-readable name for the struct that also serves as a nominal type
     pub name: StructName,
+    /// Kind constraints of the type parameters
+    pub kind_constraints: KindConstraints,
```

Comment:
nit `type_parameters`
`kind_constraints` is not informative of what this information is